src/aggregation/agg_req_with_accessor.rs

```diff
@@ -141,15 +149,15 @@ pub(crate) fn get_aggs_with_segment_accessor_and_validate(
 ) -> crate::Result<AggregationsWithAccessor> {
     let mut aggss = Vec::new();
```

Comment:
Is this a good candidate for a call to `reserve`?

Reply:
no, the cardinality is close to the user request

Reply:
But isn't it already known here because the size of `aggss` will usually match that of `aggs` or at most twice that, so that it would usually help to `aggss.reserve(aggs.len())` even if it isn't exact?

(I also wonder whether `VecWithNames::from_entries` wants to use [`Iterator::unzip`](https://doc.rust-lang.org/stable/std/iter/trait.Iterator.html#method.unzip) but admittedly unrelated to this change.)

Reply:
Looking at this again, I wonder why the intermediate `aggss` is built at all instead of directly collecting into a `VecWithNames`?

Reply:
This is not performance relevant, it's only a couple of objects

Reply:
I agree that this is certainly not measurable as an individual change so I agree with your point that this is certainly not a necessary change. I would like to add two observations though: The cost of small memory allocations is usually not dominated by their size, but by the constant cost of the synchronization necessary to call into the allocator implementation (which is why thread-local caches help so much in the average case). There is also a notion called "distributed fat" which suggests that even though these changes are not measurable individually, they can add up to measurable costs when considered in aggregate.

Of course, this is not really a good reason to fight every single allocation. Only, to avoid them if there is no loss in readability and functionality implied by avoiding them. Personally, I see calling `reserve` (or better yet using `Extend` or `FromIterator` which does this automatically) in this category but this admittedly more of a style question.